src/firefly/java/edu/caltech/ipac/firefly/server/persistence/IbeFileRetrieve.java

```diff
@@ -65,12 +64,22 @@ private List<RelatedData> findRelatedDataList(ServerRequest r) {
         String mission = r.getParam(MISSION);
         WorldPt wp= getWorldPtFromCenterParam(r.getParam("center"));
         String subsize= r.getParam("subsize");
-        if (wp==null || mission==null || subsize==null) return null;
 
         switch (mission.toLowerCase()) {
             case "wise":
-                return IbeQueryArtifact.getWiseRelatedData(wp, subsize,r.getParam("band"));
+                String scanId = r.getParam("scan_id");
+                String frameNum = r.getParam("frame_num");
+                String coaddId = r.getParam("coaddid");
+                if ( mission==null || subsize==null) return null;
+                if (wp != null) {
+                    return IbeQueryArtifact.getWiseRelatedData(wp, subsize, r.getParam("band"));
+                } else if (wp==null && scanId !=null) {
+                    return IbeQueryArtifact.getWiseScanIdRelatedData(scanId, subsize, r.getParam("band"), frameNum);
+                }  else if (wp==null && coaddId !=null) {
+                    return IbeQueryArtifact.getWiseCoaddIdRelatedData(coaddId, subsize, r.getParam("band"));
+                }
             case "2mass":
+                if (wp==null || mission==null || subsize==null) return null;
```

Comment:
line 73 already check for mission and subsize so you just need 
```java
if (wp==null) return null;
```

Reply:
> line 73 already check for mission and subsize so you just need
> 
> ```java
> if (wp==null) return null;
> ```

the check was done in switch case `wise`, should be done in `2mass` case again?